include/unitTestMain.h

```diff
@@ -21,13 +21,42 @@
 #include <cppunit/TestResultCollector.h>
 #include <cppunit/TestRunner.h>
 
+/// implements test program's main() function while enabling customization
+class TestProgram
+{
+public:
+    virtual ~TestProgram() = default;
+
+    /// Runs before all tests.
+    /// Does nothing by default.
+    virtual void startup() {}
+
+    /// Implements main(), combining all the steps.
+    /// Must be called from main().
+    /// \returns desired main() result.
+    int run(int argc, char *argv[]);
+
+private:
+    bool runTests();
+};
+
 int
-main(int, char *[])
```

Comment:
With main() removal, we should rename this header file. I think that is best done in the renaming-only PR that will also rename many of the test case files. We have used this noise-reduction strategy in recent commit 41e9c9f as well.

Reply:
I only agreed with that delay because you indicated that there was major followup work that a delay would help simplify. I assume this PR is that followup change.

FTR; I am accepting this delayed rename for these as well on the assumption that the two large sets of `src/tests/` files needing rename are not exactly identical. Please do submit the rename PR ASAP though, having separate renames does make code archaeology harder with how git does renaming.

Reply:
> I only agreed with that delay because you indicated that there was major followup work that a delay would help simplify. I assume this PR is that followup change.

I cannot find the indications you are thinking about. I [said](https://github.com/squid-cache/squid/pull/1288#pullrequestreview-1312473019) the following regarding delaying renames in recent commit https://github.com/squid-cache/squid/commit/41e9c9f078b45aac6ad49fa93020a862bc278b9b, but that does not match your recollection above: "The only large known change that I left out of scope is renaming the corresponding source files: That work simply requires more time than I have now. IMO, we do not have to rename in this PR because this PR represents reasonably scoped progress, even without the file name changes."

> Please do submit the rename PR ASAP though

I was hoping that Francesco will do that renaming so that I can focus on other things. This PR changes require renaming a single file -- unitTestMain.h.  Commit https://github.com/squid-cache/squid/commit/41e9c9f078b45aac6ad49fa93020a862bc278b9b changes (and inconsistencies in existing official code) require renaming dozens of files. Said that, if I am made responsible for solving this problem as well, then I _will_ rename them all. Just let me know when you decide that I should be the one to do it.

@kinkie, do you plan to post a renaming map and then apply reviewed renaming changes? If you plan to work on that, do you need my help?